matlab/test/arrow/type/tTime32Type.m

```diff
@@ -0,0 +1,145 @@
+                testCase.verifyEqual(type.TimeUnit, unit);
+            end
+        end
+
+        function SupplyTimeUnitString(testCase)
+            % Supply TimeUnit as a string value. Verify TimeUnit is set to
+            % the appropriate TimeUnit enum value.
+            import arrow.type.*
+            unitString = ["second", "millisecond"];
+            expectedUnit = [TimeUnit.Second, TimeUnit.Millisecond];
+
+            for ii = 1:numel(unitString)
+                type = testCase.ConstructionFcn(TimeUnit=unitString(ii));
+                testCase.verifyEqual(type.TimeUnit, expectedUnit(ii));
+            end
+        end
+
+        function ErrorIfAmbiguousTimeUnit(testCase)
+            % Verify that an error is thrown if an ambiguous value is
+            % provided for the TimeUnit name-value pair.
+            fcn = @() testCase.ConstructionFcn(TimeUnit="mi");
+            testCase.verifyError(fcn, "MATLAB:validation:UnableToConvert");
+        end
+
+        function ErrorIfTimeUnitIsNonScalar(testCase)
+            % Verify that an error is thrown if a nonscalar value is
+            % provided for the TimeUnit name-value pair.
+            units = [arrow.type.TimeUnit.Second; arrow.type.TimeUnit.Millisecond];
+            fcn = @() testCase.ConstructionFcn(TimeUnit=units);
+            testCase.verifyError(fcn, "MATLAB:validation:IncompatibleSize");
+
+            units = ["second" "millisecond"];
+            fcn = @() testCase.ConstructionFcn(TimeUnit=units);
+            testCase.verifyError(fcn, "MATLAB:validation:IncompatibleSize");
+        end
+
+        function Display(testCase)
+            % Verify the display of Time32Type objects.
+            %
+            % Example:
+            %
+            %  Time32Type with properties:
+            %
+            %          ID: Time32
+            %    TimeUnit: Second
+            %
+            type = testCase.ConstructionFcn(TimeUnit="Second"); %#ok<NASGU>
+            classnameLink = "<a href=""matlab:helpPopup arrow.type.Time32Type"" style=""font-weight:bold"">Time32Type</a>";
+            header = "  " + classnameLink + " with properties:" + newline;
+            body = strjust(pad(["ID:"; "TimeUnit:"]));
+            body = body + " " + ["Time32"; "Second"];
+            body = "    " + body;
+            footer = string(newline);
+            expectedDisplay = char(strjoin([header body' footer], newline));
+            actualDisplay = evalc('disp(type)');
+            testCase.verifyEqual(actualDisplay, expectedDisplay);
+        end
+
+        function TimeUnitNoSetter(testCase)
+            % Verify that an error is thrown when trying to set the value
+            % of the TimeUnit property.
+            schema = arrow.time32(TimeUnit="Millisecond");
+            testCase.verifyError(@() setfield(schema, "TimeUnit", "Second"), "MATLAB:class:SetProhibited");
+        end
+
+        function BitWidthNoSetter(testCase)
```

Comment:
Can you create an issue to add `BitWidthNoSetter`, `IDNoSetter`,  and`NumFieldsNoSetter` test cases to the parent test class?

Reply:
That's a good point. Created issue #37253.

Reply:
Marking this as resolved.